PrismaDB-QueryAST/DML/Expressions/BooleanExpressions.cs

```diff
@@ -33,6 +34,98 @@ public override bool Equals(object other) { return (Alias.Equals((other as Boole
         public override int GetHashCode() { return unchecked(Alias.GetHashCode() * (NOT.GetHashCode() + 1)); }
     }
 
+    public class BooleanLike : BooleanExpression
+    {
+        public ColumnRef Column;
+        public StringConstant SearchValue;
+
+        public BooleanLike()
+        {
+            setValue(new ColumnRef(""), new StringConstant(), false);
+        }
+
+        public BooleanLike(ColumnRef column, StringConstant value)
+            : this()
+        {
+            setValue((ColumnRef)column.Clone(), value);
+        }
+
+        public BooleanLike(ColumnRef column, StringConstant value, bool NOT)
+            : this(column, value)
+        {
+            setValue((ColumnRef)column.Clone(), value, NOT);
+        }
+
+        public override void setValue(params object[] value)
+        {
+            if (value.Length == 0)
+                throw new ArgumentException("BooleanLike.setValue expects at least one argument");
+            else if (value.Length == 1)
+                NOT = (bool)value[0];
+            else if (value.Length == 2)
+            {
+                Column = (ColumnRef)value[0];
+                SearchValue = (StringConstant)value[1];
+            }
+            else
+            {
+                Column = (ColumnRef)value[0];
+                SearchValue = (StringConstant)value[1];
+                NOT = (bool)value[2];
+            }
+        }
+
+        public override object Clone()
+        {
+            var column_clone = Column.Clone() as ColumnRef;
+            var svalue_clone = SearchValue.Clone() as StringConstant;
+
+            var clone = new BooleanLike(column_clone, svalue_clone, NOT);
+
+            return clone;
+        }
+
+        public override object Eval(ResultRow r)
```

Comment:
This `Eval` would not work in a general case. Imagine `ab\_de` it will be transformed into `ab\.de`, which only matches `"ab.de"`.

Reply:
I have constructed an algorithm to do wild card search for both % and _. Both wild cards have been heavily tested. Unit test is also included. Please review.